test/AutoDiff/validation-test/forward_mode.swift

```diff
@@ -1319,4 +1319,21 @@ ForwardModeTests.test("ForceUnwrapping") {
   expectEqual(5, forceUnwrap(Float(2)))
 }
 
+//===----------------------------------------------------------------------===//
+// Array methods from ArrayDifferentiation.swift
+//===----------------------------------------------------------------------===//
+ForwardModeTests.test("Array.differentiableMap") {
+  let a: Array<Float> = [1, 2, 3]
+
+  func multiplyMap(a: Array<Float>) -> Array<Float> {
```

Comment:
Though it doesn't really matter here, please omit argument labels for `multiplyMap` and `squareMap`.

Argument labels are significant in Swift and affect call-site usage: it's more natural to call `multiplyMap(array)` instead of `multiplyMap(a: array)`.

```suggestion
  func multiplyMap(_ a: [Float]) -> [Float] {
```